packages/aws-rfdk/lib/deadline/lib/thinkbox-docker-images.ts

```diff
@@ -0,0 +1,183 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { randomBytes } from 'crypto';
+import * as path from 'path';
+
+import {
+  ContainerImage,
+} from '@aws-cdk/aws-ecs';
+import {
+  Code,
+  SingletonFunction,
+  Runtime,
+} from '@aws-cdk/aws-lambda';
+import { RetentionDays } from '@aws-cdk/aws-logs';
+import {
+  Construct,
+  CustomResource,
+  Duration,
+  Token,
+} from '@aws-cdk/core';
+
+import {
+  IVersion,
+  RenderQueueImages,
+  ThinkboxManagedDeadlineDockerRecipes,
+  UsageBasedLicensingImages,
+} from '.';
+
+/**
+ * Interface to specify the properties when instantiating a {@link ThinkboxDockerImages} instnace.
+ */
+export interface ThinkboxDockerImagesProps {
+  /**
+   * The Deadline version to obtain images for.
+   * @default latest
+   */
+  readonly version?: IVersion
+}
+
+/**
+ * An API for interacting with staged Deadline Docker images provided by AWS Thinkbox.
+ *
+ * This provides container images as required by RFDK's Deadline constructs such as
+ *
+ * * {@link @aws-rfdk/deadline#RenderQueue}
+ * * {@link @aws-rfdk/deadline#UsageBasedLicensing}
+ *
+ * @example Construct a RenderQueue
+ *
+ * import { App, Stack, Vpc } from '@aws-rfdk/core';
+ * import { RenderQueue, Repository, ThinkboxDockerRecipes } from '@aws-rfdk/deadline';
+ * const app = new App();
+ * const stack = new Stack(app, 'Stack');
+ * const vpc = new Vpc(app, stack);
+ * const images = new ThinkboxDockerImages(stack, 'Image');
+ * const repository = new Repository(stack, 'Repository', {
+ *   vpc,
+ *   version: recipes.version
+ * });
+ *
+ * const renderQueue = new RenderQueue(stack, 'RenderQueue', {
+ *   images: images.forRenderQueue(),
+ *   // ...
+ * });
+ */
+export class ThinkboxDockerImages extends Construct {
+  /**
+   * A {@link DockerImageAsset} that can be used to build Thinkbox's Deadline RCS Docker Recipe into a
+   * container image that can be deployed in CDK.
+   *
+   * @param scope The parent scope
+   * @param id The construct ID
+   */
+  public readonly remoteConnectionServer: ContainerImage;
+
+  /**
+   * A {@link DockerImageAsset} that can be used to build Thinkbox's Deadline License Forwarder Docker Recipe into a
+   * container image that can be deployed in CDK.
+   *
+   * @param scope The parent scope
+   * @param id The construct ID
+   */
+  public readonly licenseForwarder: ContainerImage;
+
+  /**
+   * The version of Deadline in the stage directory.
```

Comment:
Yes, when its undefined, the image will attach no version specifier suffix to the image tag. In that case, ECS will default to the `:latest` specifier.

I've removed this from being a public property since consumers would need to code defensively with a null/undefined check. If they need the version, they can still pass it around in the CDK app externally. What do you think?

Reply:
This looks good. Thanks Josh!